tensorflow_federated/python/core/utils/execution_tracing.py

```diff
@@ -0,0 +1,313 @@
+import asyncio
+import contextlib
+from enum import Enum
+import functools
+import inspect
+import logging
+import re
+
+from tensorflow.python.framework.ops import EagerTensor as tf_EagerTensor
+
+from tensorflow_federated.proto.v0 import computation_pb2
+from tensorflow_federated.python.common_libs import anonymous_tuple
+from tensorflow_federated.python.common_libs import tracing
+from tensorflow_federated.python.core.api import computation_types
+from tensorflow_federated.python.core.impl import computation_impl
+from tensorflow_federated.python.core.impl.compiler import intrinsic_defs
+from tensorflow_federated.python.core.impl.executors import caching_executor
+from tensorflow_federated.python.core.impl.executors import composing_executor
+from tensorflow_federated.python.core.impl.executors import eager_tf_executor
+from tensorflow_federated.python.core.impl.executors import federating_executor
+from tensorflow_federated.python.core.impl.executors import reference_resolving_executor
+from tensorflow_federated.python.core.impl.executors import remote_executor
+from tensorflow_federated.python.core.impl.executors import sizing_executor
+from tensorflow_federated.python.core.impl.executors import thread_delegating_executor
+from tensorflow_federated.python.core.impl.executors import transforming_executor
```

Comment:
These dependencies will require that this tracing utility have a privileged visibility. I think where it is located is probably the appropriate location given the dependencies. But let me ask, is it possible to remove these dependencies? It seems to me like you are simply stripping the module name from all non-primative types? I think we could do this without having an inclusion list of types that we check against. Additionally, we would not have to come back and add more when we add executors.

Reply:
That's a good point -- this code grew out of some hastier monkey-patching code, which used required the actual types. I'll remove them.

On the other hand, I'm trying to think through how we would differentiate between, for example, an executor and an executor value without building up inclusion lists. One possibility would be matching on the type name, but that doesn't feel robust. Do you have any alternatives in mind?

Reply:
I'll comment inline in the iteration you pushed.